include/mneme/storage.hpp

```diff
@@ -20,15 +21,26 @@ namespace detail {
 template <DataLayout TDataLayout, typename... Ids> struct DataLayoutAllocatePolicy;
 
 template <DataLayout TDataLayout, std::size_t Extent, typename... Ids>
-struct DataLayoutAccessPolicy;
+struct DataLayoutAccessPolicy {
+    static constexpr auto layout = TDataLayout;
+};
 
 template <typename... Ids> struct DataLayoutAllocatePolicy<DataLayout::AoS, Ids...> {
     using type = tagged_tuple<Ids...>*;
 
     constexpr static void allocate(type& c, std::size_t size) {
-        c = new tagged_tuple<Ids...>[size];
+        static_assert(allSameAllocator<Ids...>(),
+                      "AoS layout only works if all Ids share the same allocator.");
+        if constexpr (AllocatorInfo<Ids...>::template allSameAllocatorAs<AlignedAllocatorBase>()) {
+            constexpr auto alignment = getMaxAlignment<Ids...>();
+            using allocator_t = AlignedAllocator<tagged_tuple<Ids...>, alignment>;
+            auto allocator = allocator_t();
+            c = std::allocator_traits<allocator_t>::allocate(allocator, size);
+        } else {
+            c = new tagged_tuple<Ids...>[size];
+        }
     }
-    constexpr static void deallocate(type& c) { delete[] c; }
+    constexpr static void deallocate(type& c, std::size_t) { delete[] c; }
```

Comment:
Yes, and no. As far as I know, most allocators are stateless.
See e.g. [here](https://github.com/SeisSol/SeisSol/blob/2f4942e8de21aebd56e179eb11837acd40ae4c07/src/Initializer/MemoryAllocator.cpp#L114), how we do it in SeisSol currently.

I thought that it was required that all custom allocators have to be stateless, but that was changed in C++11:

>  Custom allocators may contain state. Each container or another allocator-aware object stores an instance of the supplied allocator and controls allocator replacement through std::allocator_traits. 

So it might make sense in the future to store the allocator. Right now even the size parameter is unused for deallocate.

Reply:
Ah, now I understood your comment. I'm now using the correct deallocating method.